lib/hanami/container.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+
+require "dry/system/container"
+
+module Hanami
+  # Hanami private IoC
+  #
+  # @since 2.0.0
+  class Container < Dry::System::Container
+    boot(:root) do
+      start do
+        register(:root, Hanami.root)
+      end
+    end
```

Comment:
This is just a configuration, `root` is not a component. What's the reason for treating it as a component? Furthermore, dry-system's containers have `config.root` which you want to set correctly.

Reply:
I changed it to use the appropriate container setting: https://github.com/hanami/hanami/pull/975/commits/cfd7adbc2d996bc076dcec9720fdf50912c78929